packages/components/pie-toast/test/component/pie-toast.spec.ts

```diff
@@ -1,317 +1,256 @@
-const messageSelector = `[data-test-id="${rootSelector}-message"]`;
-const closeSelector = `[data-test-id="${rootSelector}-close"]`;
-const footerSelector = `[data-test-id="${rootSelector}-footer"]`;
-const leadingActionSelector = `[data-test-id="${rootSelector}-leading-action"]`;
-
-/**
- * A small amount of time in milliseconds to wait in order to allow the component to have rendered and its animations to have completed.
- */
-const COMPONENT_RENDER_DURATION_MS = 250;
-
-function getToastIconSelectors (variant: string): string {
-    return `[data-test-id="${rootSelector}-heading-icon-${variant}"]`;
-}
-
 test.describe('PieToast - Component tests', () => {
-    // IMPORTANT: Mounting and Unmounting the component before each test ensures that any tests that do not explicitly
-    // mount the component will still have it available in Playwright's cache (loaded and registered in the test browser)
-    test.beforeEach(async ({ mount }) => {
-        const component = await mount(PieToast);
-        await component.unmount();
-    });
-
     const mainAction = {
         text: 'Confirm',
         ariaLabel: 'Button that confirm the action',
     };
 
-    test('should render successfully', async ({ mount, page }) => {
+    test('should render successfully', async ({ page }) => {
         // Arrange
-        await mount(PieToast, {
-            props: {} as ToastProps,
-        });
-
-        // Wait for the component to render alongside its animations
-        await page.waitForTimeout(COMPONENT_RENDER_DURATION_MS);
+        const toastPage = new BasePage(page, 'toast');
+        await toastPage.load();
 
         // Act
-        const toast = page.locator(componentSelector);
+        const toastComponent = page.getByTestId(toast.selectors.container.dataTestId);
 
         // Assert
-        expect(toast).toBeVisible();
+        await expect(toastComponent).toBeVisible();
     });
 
     test.describe('Props', () => {
         test.describe('isOpen', () => {
-            test('should have c-toast--animate-out class if isOpen is false', async ({ mount, page }) => {
+            test('should have c-toast--animate-out class if isOpen is false', async ({ page }) => {
                 // Arrange
-                await mount(PieToast, {
-                    props: {
-                        isOpen: false,
-                    } as ToastProps,
-                });
+                const toastPage = new BasePage(page, 'toast');
+                const props: ToastProps = {
+                    isOpen: false,
+                    duration: 250,
+                };
+                await toastPage.load({ ...props });
 
-                // Wait for the component to render alongside its animations
-                await page.waitForTimeout(COMPONENT_RENDER_DURATION_MS);
+                //  Wait for the component to render and close alongside its animations
+                await page.waitForTimeout(250);
```

Comment:
Can we try this instead?

await expect.soft(page.getByTestId(toast.selectors.container.dataTestId)).toBeVisible();